<?xml version="1.0"?>
<launch>

  <arg name="hiros_skeleton_tracker_node_required" default="false" />
  <arg name="node_name"                            default="node"  />

  <!--************************************-->
  <!-- Hi-Ros Skeleton Tracker Parameters -->
  <!--************************************-->
  <arg name="in_skeleton_group_topic" default=""      />
  <arg name="out_msg_topic_name"      default="track" />

  <arg name="min_keypoints"           default="0"     />
  <arg name="max_distance"            default="-1"    />
  <arg name="max_delta_t"             default="0"     />

  <!--*******************************-->
  <!-- Hi-Ros OpenPose Tracking Node -->
  <!--*******************************-->
  <node pkg="hiros_skeleton_tracker" type="hiros_skeleton_tracker" ns="hiros/st" name="$(arg node_name)" output="screen" required="$(arg hiros_skeleton_tracker_node_required)">
    <param name="node_name"               type="str"    value="$(arg node_name)"               />

    <param name="in_skeleton_group_topic" type="str"    value="$(arg in_skeleton_group_topic)" />
    <param name="out_msg_topic_name"      type="str"    value="$(arg out_msg_topic_name)"      />

    <param name="min_keypoints"           type="int"    value="$(arg min_keypoints)"           />
    <param name="max_distance"            type="double" value="$(arg max_distance)"            />
    <param name="max_delta_t"             type="double" value="$(arg max_delta_t)"             />
  </node>

</launch>
